tests/phpunit/integration/Modules/Reader_Revenue_ManagerTest.php

```diff
@@ -73,4 +89,76 @@ public function test_service_classes_exist() {
 			class_exists( 'Google\Site_Kit_Dependencies\Google\Service\SubscribewithGoogle' )
 		);
 	}
+
+	public function test_get_datapoints() {
+		$this->assertEqualSets(
+			array(
+				'publications',
+			),
+			$this->reader_revenue_manager->get_datapoints()
+		);
+	}
+
+	public function test_get_publications() {
+		FakeHttp::fake_google_http_handler(
+			$this->reader_revenue_manager->get_client(),
+			function ( Request $request ) {
+				$url = parse_url( $request->getUri() );
+
+				switch ( $url['path'] ) {
+					case '/v1/publications':
+						$publications = array(
+							array(
+								'publicationId'         => 'ABCDEFGH',
```

Comment:
I find it's best to create the response using model instances, that way it's less likely to be able to return an invalid shape as we may be here.